pkg/server/plugin/nodeattestor/awsiid/organization.go

```diff
@@ -0,0 +1,264 @@
+func (o *organizationValidation) ValidateAccountBelongstoOrg(ctx context.Context, orgClient organizations.ListAccountsAPIClient, accoundIDofNode string) (bool, error) {
+
+	orgAccountList, isStale, err := o.checkIfOrgAccountListIsStale(ctx)
+	if err != nil {
+		return false, err
+	}
+	// refresh the account map
+	if isStale {
+		orgAccountList, err = o.fetchAccountsListFromOrg(ctx, orgClient, false)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	// Check if account contains in organization list.
+	// If this return false we will check `X` number of times in that window, to avoid
+	// genuine accounts which are newly added
+	exist := checkIfAccountIdExist(orgAccountList, accoundIDofNode)
+	if !exist {
+		remRetries := o.getRetries()
+		if remRetries > 0 && !isStale {
+			orgAccountList, err = o.fetchAccountsListFromOrg(ctx, orgClient, true)
+			if err != nil {
+				return false, err
+			}
+			exist = checkIfAccountIdExist(orgAccountList, accoundIDofNode)
+			return exist, nil
+		}
+	}
+
+	return exist, nil
+}
+
+// Check if the org account list is stale.
+func (o *organizationValidation) checkIfOrgAccountListIsStale(ctx context.Context) (map[string]bool, bool, error) {
+	o.mutex.RLock()
+	defer o.mutex.RUnlock()
+
+	// Map is empty that means this is first time plugin is being initialised
+	if len(o.orgListAccountMap) == 0 {
+		return nil, true, nil
+	}
+
+	// Get the timestamp from config
+	existingTimestamp := o.orgListAccountMapCreationTime
+
+	currTimeStamp := time.Now().UTC()
+
+	// Check diff of timestamp of org acc map & current time if its more than ttl, refresh the list
+	if currTimeStamp.Sub(existingTimestamp) >= time.Duration(o.orgAccountListCacheTTL) {
+		return nil, true, nil
+	}
+
+	return o.orgListAccountMap, false, nil
+}
+
+// Get the list of accounts belonging to organization and catch them
+func (o *organizationValidation) fetchAccountsListFromOrg(ctx context.Context, orgClient organizations.ListAccountsAPIClient, catchBurst bool) (map[string]bool, error) {
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+
+	// Make sure : map is not updated from different go routine. This will make sure we dont make extra calls
+	// * if we make this call before ttl expires, we are doing retry/catchburst that means we need to bypass this validation
+	if !catchBurst && len(o.orgListAccountMap) != 0 && time.Now().UTC().Sub(o.orgListAccountMapCreationTime) <= time.Duration(o.orgAccountListCacheTTL) {
+		return o.orgListAccountMap, nil
+	}
+
+	// Avoid if other thread has already updated the map
+	if catchBurst && o.retries == 0 {
+		return o.orgListAccountMap, nil
+	}
+
+	// Get the list of accounts
+	listAccountsOp, err := orgClient.ListAccounts(ctx, &organizations.ListAccountsInput{})
+	if err != nil {
+		return nil, fmt.Errorf("issue while getting list of accounts: %v", err)
+	}
+
+	//Build new org accounts list
+	orgAccountsMap := make(map[string]bool)
+
+	// Update the org account list cache with ACTIVE accounts & handle pagination
```

Comment:
What are the other possible states?

Reply:
Could be suspended or pending closure. More can be found [here](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/organizations@v1.23.5/types#AccountStatus). Tried to capture this in test cases as well.